pkg/infra/baseMessage.go

```diff
@@ -0,0 +1,72 @@
+package infra
+
+//Define message types to enable basic protocols, voting systems ...etc
+
+type MessageType int
+
+// woonmoon(TODO): See if there's a more Go-friendly way to do this...
+// ideally Ask/State/Request would be three different Messagetypes
```

Comment:
I think this is currently good, I don't see a reason to have further inheritance for these three types (Ask / State / Request) unless there is a functional difference between them that justifies the inheritance